### DISCUSSION TITLE
fluent bit plug-in SQL Server Log UTF16

### DISCUSSION THREAD

**Discussion by sairnuth2** [fluent/fluent-bit#8411] [General]
I feel SQL Server error log is in UTF 16 format. I am unable to filter the content from that log due to the format.  Could you please help me to read the SQL Error log file with specific error messages.  Or if you have any way of convert UTF 16 format file. Thank you.

**Comment by sairnuth2** [comment]
[fluent-sql.zip](https://github.com/fluent/fluent-bit/files/14029175/fluent-sql.zip)
Attaching my current conf and sample SQL error log files

**Comment by sairnuth2** [comment]
any suggestions for my request?

**Reply by cosmo0920** [reply]
Hi, we're tying to solve this issue in #9468.
This could be effective for UTF-16LE/UTF-16BE encoded logs.
However, as far as I tried with that branch, I experienced that the patch can consume with additional parameters which is `Unicode.Encoding`. Not sure why the parameter works with `Unicode.Encoding UTF-16BE` for your reproducible. Plus, your reproducible is able to start to work with `Regex message Login.*failed.*` in grep plugin settings.

**Reply by cosmo0920** [reply]
I updated [the above PR](https://github.com/fluent/fluent-bit/pull/9468) and started to process your reproducible with UTF-16LE.
Your log can be consumed as UTF-16LE with `Unicode.Encoding UTF-16LE` parameter and also grep filter plugin can filter out not matched log lines.
